pkgs/desktops/lomiri/qml/lomiri-action-api/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, gitUpdater
+, testers
+, cmake
+, dbus
+, dbus-test-runner
+, pkg-config
+, qtbase
+, qtdeclarative
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "lomiri-action-api";
+  version = "1.1.2";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/lomiri-action-api";
+    rev = finalAttrs.version;
+    hash = "sha256-FOHjZ5F4IkjSn/SpZEz25CbTR/gaK4D7BRxDVSDuAl8=";
+  };
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  patches = [
+    ./0001-Drop-deprecated-qt5_use_modules.patch
```

Comment:
Can this be upstreamed?

Reply:
I wrote abunch of `qt5_use_modules` patches in these packages, and they're all upstreamable (and should be upstreamed).

<sub>I know I should do this ahead of reviews, but it's kinda hard to find the motivation for that when it might take several months to find PR reviewers/mergers in Nixpkgs... Big thanks at this point for going through these btw.</sub>

Will do this right now.

Reply:
Submitted upstream & `fetchpatch`ed, along with `finalAttrs.doCheck` -> `finalAttrs.finalPackage.doCheck` and a reference to the Qt5 `qdoc` PR.

Reply:
`finalAttrs.doCheck` -> `finalAttrs.finalPackage.doCheck` is not really needed in your case, since you do the cross logic by setting `doCheck`. I figured a quick treewide fix later is the way to go

Reply:
I would agree, but another reviewer demanded it on another of these package. Applied it just to avoid further complaints.